### PR TITLE
ARROW-7720: [C++][Python] Add check_metadata argument to Table.equals

### PR DESCRIPTION
**Pull request by kszucs**
Use 

```py
table.equals(other, check_metadata=False)
``` 

instead of

```py
table.replace_schema_metadata().equals(other)
```

### CODE REVIEW

**Comment by kszucs** [comment]
Don't merge it yet, need to add tests.

**Comment by jorisvandenbossche** [comment]
+1, that's also something I thought we should add while writing those tests with `replace_schema_metadata()`

File: python/pyarrow/table.pxi [hunk 1]
```diff
@@ -1058,7 +1058,7 @@ cdef class Table(_PandasConvertible):
         except TypeError:
             return NotImplemented
 
-    def equals(self, Table other):
+    def equals(self, Table other, bint check_metadata=True):
```
Comment (pitrou) [inline python/pyarrow/table.pxi]: You should add the new parameter in the docstring below.

**Review by pitrou — CHANGES_REQUESTED** [review]
+1, just a missing docstring update

**Review by nealrichardson — CHANGES_REQUESTED** [review]
Care to add this to the R bindings too? :) Should be as trivial as the Python change.

**Comment by kszucs** [comment]
> Care to add this to the R bindings too? :) Should be as trivial as the Python change.

Sure.

**Review by pitrou — APPROVED** [review]

**Review by xhochy — APPROVED** [review]
+1, cool!

File: r/src/table.cpp [hunk 2]
```diff
@@ -88,7 +88,7 @@ std::shared_ptr<arrow::Table> Table__Slice2(const std::shared_ptr<arrow::Table>&
 
 // [[arrow::export]]
 bool Table__Equals(const std::shared_ptr<arrow::Table>& lhs,
-                   const std::shared_ptr<arrow::Table>& rhs) {
+                   const std::shared_ptr<arrow::Table>& rhs, bool check_metadata) {
```
Comment (nealrichardson) [inline r/src/table.cpp]: I think you need to pass `check_metadata` on the next line.

**Review by nealrichardson — CHANGES_REQUESTED** [review]
Would be great to add a test in R; might have caught the issue I flagged.

**Comment by kszucs** [comment]
I've also added the R bindings locally, just forgot to push... Thanks @pitrou!

File: r/src/table.cpp (hunk 2 above)
Comment (pitrou) [reply r/src/table.cpp]: Uh, what do you mean? All CI passed (except for an unrelated AppVeyor issue).

**Comment by pitrou** [comment]
> Would be great to add a test in R; might have caught the issue I flagged.

Which issue is that?

**Comment by pitrou** [comment]
In any case, there is no metadata test currently in R (not even metadata construction), so I wouldn't know how to do that. I'll merge here :-)

File: r/src/table.cpp (hunk 2 above)
Comment (nealrichardson) [reply r/src/table.cpp]: Exactly my point about the test. We're accepting this new argument but not passing it to the C++ function, so this is not really implemented, and we would have known that if we had a test.

Comment (pitrou) [reply r/src/table.cpp]: Ahah, sorry, I had misunderstood you. Hmm, can I let you open a JIRA and submit a PR?

Comment (nealrichardson) [reply r/src/table.cpp]: https://issues.apache.org/jira/browse/ARROW-7786

**Pull request closed without merging**